### PR TITLE
Add library scan & refresh indicators

### PR DESCRIPTION
**Pull request by MrTimscampi**
Add library refresh indicators:
https://user-images.githubusercontent.com/19396809/102949825-7eb77680-44c9-11eb-934f-dd30805fff6e.mp4

(WIP) Add the possibility to trigger library scans

### CODE REVIEW

File: components/Card.vue [hunk 1]
```diff
@@ -230,6 +238,17 @@ export default Vue.extend({
...
+  created() {
+    this.$store.subscribe((mutation, state) => {
+      if (
+        mutation.type === 'SOCKET_ONMESSAGE' &&
+        state.socket.message.MessageType === 'RefreshProgress' &&
+        state.socket.message.Data.ItemId === this.item.Id
+      ) {
+        this.refreshProgress = state.socket.message.Data.Progress;
+      }
+    });
```
Comment (Camc314) [inline components/Card.vue:251]: I know this is only a draft, but it is probably better to check the card type before subscribing to the websocket messages

Comment (MrTimscampi) [reply components/Card.vue:251]: The idea here is that other items could also get refresh progress down the line.

There are bits in the server about provider refresh progress which, while unused now, could be used in the future to inform the user about individual item refreshes.

However, for now, we could limit it to libraries.

**Comment by ferferga** [comment]
@MrTimscampi what's left for this PR so it's ready to be reviewed?

**Comment by MrTimscampi** [comment]
> @MrTimscampi what's left for this PR so it's ready to be reviewed?

It depends if we want to add the library refresh in this or make a new PR. Currently, it only has the indicators for notifying of a refresh.

**Comment by ferferga** [comment]
@MrTimscampi I would say it's fine to get this done right now, fewer changes to review later.

**Review by Camc314 — APPROVED** [review]

**Pull request merged**